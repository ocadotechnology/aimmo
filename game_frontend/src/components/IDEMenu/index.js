import styled from 'styled-components'
import React, { Component } from 'react'
import PropTypes from 'prop-types'

const IDEMenuLayout = styled.nav`
  background-color: pink
  grid-area: ide-menu
`

export default class IDEMenu extends Component {
  render () {
    return (
      <IDEMenuLayout>
        <button
          id='get-code-button'
          onClick={this.props.getCode} >Get Code</button>
        <button
          id='post-code-button'
          onClick={this.props.postCode} >Post Code</button>
      </IDEMenuLayout>
    )
  }
}

IDEMenu.propTypes = {
  getCode: PropTypes.func,
  postCode: PropTypes.func
}
