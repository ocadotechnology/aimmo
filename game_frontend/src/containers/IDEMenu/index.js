import styled from 'styled-components'
import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { actions } from 'features/Editor'

export const IDEMenuLayout = styled.nav`
  background-color: ${props => props.theme.palette.background.default}
  grid-area: ide-menu
`

export class IDEMenu extends Component {
  render () {
    return (
      <IDEMenuLayout>
        <button
          id='post-code-button'
          onClick={this.props.postCode} >Post Code</button>
      </IDEMenuLayout>
    )
  }
}

IDEMenu.propTypes = {
  postCode: PropTypes.func
}

const mapStateToProps = () => ({})

const mapDispatchToProps = {
  postCode: actions.postCodeRequest
}

export default connect(mapStateToProps, mapDispatchToProps)(IDEMenu)
