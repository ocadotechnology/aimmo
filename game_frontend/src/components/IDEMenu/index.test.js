/* eslint-env jest */
import React from 'react'
import IDEMenu, { IDEMenuLayout } from 'components/IDEMenu'
import { shallow } from 'enzyme'
import createShallowWithTheme from 'testHelpers/createShallow'

describe('<IDEMenu />', () => {
  it('renders correctly', () => {
    const tree = createShallowWithTheme(<IDEMenu />)
    expect(tree).toMatchSnapshot()
  })

  it('calls the postCode function in props when Post code button is pressed', () => {
    const postCode = jest.fn()
    const props = {
      postCode
    }

    const component = shallow(<IDEMenu {...props} />)

    component.find('#post-code-button').simulate('click')
    expect(postCode).toBeCalled()
  })
})

describe('<IDEMenuLayout />', () => {
  it('renders correctly', () => {
    const tree = createShallowWithTheme(<IDEMenuLayout />)
    expect(tree).toMatchSnapshot()
  })
})
