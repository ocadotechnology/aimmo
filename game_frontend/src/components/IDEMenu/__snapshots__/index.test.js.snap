// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IDEMenu /> renders correctly 1`] = `
<styled.nav>
  <button
    id="post-code-button"
  >
    Post Code
  </button>
</styled.nav>
`;

exports[`<IDEMenuLayout /> renders correctly 1`] = `
.c0 {
  background-color: #303030;
  grid-area: ide-menu;
}

<nav
  className="c0"
/>
`;
